src/scripts/clipperUI/componentBase.ts

```diff
@@ -115,42 +124,67 @@ export abstract class ComponentBase<TState, TProps> {
+						let posInSet = parseInt(element.getAttribute("aria-posinset"), 10);
+						if (e.which === Constants.KeyCodes.home) {
+							let firstInSet = 1;
+							ComponentBase.focusOnButton(verticalArrowKeyNavSet, firstInSet, e);
+						} else if (e.which === Constants.KeyCodes.end) {
+							let lastInSet = parseInt(element.getAttribute("aria-setsize"), 10);
+							ComponentBase.focusOnButton(verticalArrowKeyNavSet, lastInSet, e);
+						}
+
+						if (verticalArrowKeyNavSet) {
+							if (e.which === Constants.KeyCodes.up) {
+								if (posInSet === 1) {
+									return;
+								}
+								let nextPosInSet = posInSet - 1;
+								ComponentBase.focusOnButton(verticalArrowKeyNavSet, nextPosInSet, e);
+							} else if (e.which === Constants.KeyCodes.down) {
+								let setSize = parseInt(element.getAttribute("aria-setsize"), 10);
+								if (posInSet === setSize) {
+									return;
+								}
+								let nextPosInSet = posInSet + 1;
+								ComponentBase.focusOnButton(verticalArrowKeyNavSet, nextPosInSet, e);
+							}
 						}
-						let nextPosInSet = posInSet - 1;
-						ComponentBase.focusOnButton(setNameForArrowKeyNav, nextPosInSet);
-					} else if (e.which === Constants.KeyCodes.down) {
-						let setSize = parseInt(element.getAttribute("aria-setsize"), 10);
-						if (posInSet === setSize) {
-							return;
+
+						if (horizontalArrowKeyNavSet) {
+							if (e.which === Constants.KeyCodes.left) {
+								if (posInSet === 1) {
+									return;
+								}
+								let nextPosInSet = posInSet - 1;
+								ComponentBase.focusOnButton(horizontalArrowKeyNavSet, nextPosInSet, e);
+							} else if (e.which === Constants.KeyCodes.right) {
+								let setSize = parseInt(element.getAttribute("aria-setsize"), 10);
+								if (posInSet === setSize) {
+									return;
+								}
+								let nextPosInSet = posInSet + 1;
+								ComponentBase.focusOnButton(horizontalArrowKeyNavSet, nextPosInSet, e);
+							}
 						}
-						let nextPosInSet = posInSet + 1;
-						ComponentBase.focusOnButton(setNameForArrowKeyNav, nextPosInSet);
-					} else if (e.which === Constants.KeyCodes.home) {
-						let firstInSet = 1;
-						ComponentBase.focusOnButton(setNameForArrowKeyNav, firstInSet);
-					} else if (e.which === Constants.KeyCodes.end) {
-						let lastInSet = parseInt(element.getAttribute("aria-setsize"), 10);
-						ComponentBase.focusOnButton(setNameForArrowKeyNav, lastInSet);
 					}
 				}
-			},
+			}
+			,
 			onmousedown: (e: MouseEvent) => {
 				let element = e.currentTarget as HTMLElement;
 				element.style.outlineStyle = "none";
 			},
 			tabIndex: tabIndex,
-			"data-setnameforarrowkeynav": setNameForArrowKeyNav
+			"data-setnameforarrowkeynav": ariaSet
 		};
 	}
 
-	private static focusOnButton(setNameForArrowKeyNav: string, posInSet: number) {
+	private static focusOnButton(setNameForArrowKeyNav: string, posInSet: number, e: KeyboardEvent) {
+		e.preventDefault();
```

Comment:
It feels odd to pass in the KeyboardEvent into a method whose responsibility is to just focus on a button